### PR TITLE
[Feature] YOLO dataloader

### LINKED ISSUE #29: [Feature] Dataloader for YOLO data format
# Prerequisites

Please answer the following question for yourself before submitting an issue.

- [x] I checked to make sure that this feature has not been requested already.

## 1. The entire URL of the file you are using

https://github.com/tensorflow/models/tree/master/official/vision/beta

## 2. Describe the feature you request

Add decoder and parser for yolo data format.

## 3. Additional context

Add any other context about the feature request here.
Required to use SUTD's labelled data in yolo format.

## 4. Are you willing to contribute it? (Yes or No)

Yes

### PR DESCRIPTION
**Pull request by rehohoho**
## Description
What feature/issue was addressed?
Added YOLO dataloader for YOLO tasks.
Decoder parses tf.Example (protobuf format) into relevant tensors.
Parser performs augmentation and does the yolo op to generate multi-scaled bboxes and labels.

I'll leave the evaluation portion and zoom augmentation to another PR when training is done proper.

How was it resolved/added?

Any dependencies required for the change?

## Issue reference

Please reference the issue this PR will close: #29.

## Type of change

- [ ] Bug fix (non-breaking change which fixes an issue)
- [x] New feature (non-breaking change which adds functionality)
- [ ] Breaking change (fix or feature that would cause existing functionality to not work as expected)
- [ ] Documentation update
- [ ] Other (Specify)

## Tests

Any reproducable method of testing to verify changes? \
List relevant details for the test configuration.

## Checklist
- [x] I have performed a self-review of my own code. Feel free to destroy my PR
- [x] I have commented my code, particularly in hard-to-understand areas
- [x] My changes generate no new warnings.
- [ ] I have made corresponding changes to the documentation.
- [ ] I have added tests that prove my fix is effective or that my feature works.

### CODE REVIEW

**Review by MelvinFMQ — APPROVED** [review]
lgtm
